EIPS/eip-140.md

```diff
@@ -0,0 +1,50 @@
+## Preamble
+
+    EIP: 140
+    Title: REVERT instruction in the Ethereum Virtual Machine
+    Author: Alex Beregszaszi, Nikolai Mushegian (nikolai@nexusdev.us)
+    Type: Standard Track
+    Category: Core
+    Status: Draft
+    Created: 2017-02-06
+
+## Simple Summary
+
+The `REVERT` instruction provides a way to stop execution and revert state changes, without consuming all provided gas and with the ability to return a reason.
+
+## Abstract
+
+The `REVERT` instruction will stop execution, roll back all state changes done so far and provide a pointer to a memory section, which can be interpreted as an error code or message. While doing so, it will not consume all the remaining gas.
+
+## Motivation
+
+Currently this is not possible. There are two practical ways to revert a transaction from within a contract: running out of gas or executing an invalid instruction. Both of these options will consume all remaining gas. Additionally, reverting a transaction means that all changes, including LOGs, are lost and there is no way to convey a reason for aborting a transaction.
+
+## Specification
+
+The `REVERT` instruction is introduced at `0xfd`. Execution is aborted, considered as failed, and state changes are rolled back.
```

Comment:
After talking to @axic, it's more and more like those nonce-increments stay, but maybe it's enough to say "like other exceptional halting cases".